### PR TITLE
WIP: Add context.time into the attribute bag when Mixer gets a request.

### PR DESCRIPTION
**Pull request by ozevren**
This is WIP. If there is acceptance that this is the right-thing-to-do(™), then I'll make this a full-blown PR.

### CODE REVIEW

File: mixer/pkg/api/grpcServer.go [hunk 1]
```diff
@@ -129,6 +129,7 @@ func (s *grpcServer) Check(legacyCtx legacyContext.Context, req *mixerpb.CheckRe
 	// compatReqBag ensures that preprocessor input handles deprecated attributes gracefully.
 	compatReqBag := &compatBag{requestBag}
 	preprocResponseBag := attribute.GetMutableBag(nil)
+	addContextAttributes(preprocResponseBag)
```
Comment (mandarjog) [inline mixer/pkg/api/grpcServer.go]: Looks like context.time is not available during preprocess. It is stuffed directly into preprocresponse bag.
Perhaps add a childbag of compareqbag with this attribute and pass it as an input to preprocess.

Comment (ozevren) [reply mixer/pkg/api/grpcServer.go]: This is becoming really awkward. I've updated the code to show structure, but I really am not happy with where this is heading. We really should need 3 bags at most:
- The proto bag
- The compatibility bag
- The mutable bag for generated/context attributes.

I'll take another stab at simplifying the overall bag juggling.

Comment (mandarjog) [reply mixer/pkg/api/grpcServer.go]: Lets drop the compatibility bag, Though it is a good insertion points when we have incompatibilities in the future.

At the very least, the compatibility should be a no-op now. It should not do any  alias lookups, any more.

**Comment by douglas-reid** [comment]
any reason to not do this in an APA?  maybe one that takes no additional config?  Might simplify things (and maybe make it easier to extend later). Only catch would be that other APAs couldn't rely on having context.* attributes.

**Comment by ozevren** [comment]
I think we would want these to work in APAs.
Here is an update that "hijacks" the compatBag. I think we can make it a full-blown bag implementation (maybe put it in attributes package) and use it this way.

I think a more holistic/better solution would be to make proto-bag parent-aware (which should also help with delta-encoded proto bags) and use a context-bag as the root. Then, simply use a single mutable-bag to collect APA results before passing it in to the downstream call.

With that, we would end up with 3 bags on the call side, and 1 bag per-APA call. It is a significant surgery though.

**Pull request closed without merging**